crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1080,7 +1080,15 @@ async fn handle_resolve_plugins(
     .cell())
 }
 
-#[turbo_tasks::function]
+fn resolve_internal_boxed(
+    lookup_path: Vc<FileSystemPath>,
+    request: Vc<Request>,
+    options: Vc<ResolveOptions>,
+) -> Pin<Box<dyn Future<Output = Result<Vc<ResolveResult>>> + Send>> {
+    Box::pin(resolve_internal(lookup_path, request, options))
+}
+
+// Not a turbo-tasks function: Called too often
```

Comment:
I guess this might reduce caching, but I'm assuming you looked into that

Reply:
Yeah, it also cases a stack overflow due to recursive calls....

The effect of caching is probably something I need to measure...